code/modules/mod/mod_construction.dm

```diff
@@ -0,0 +1,171 @@
+
+		if(5)
+			if(istype(I, /obj/item/mod/construction/gauntlets)) //Construct
+				if(!user.temporarilyRemoveItemFromInventory(I))
+					return
+				playsound(src, 'sound/machines/click.ogg', 30, TRUE)
+				to_chat(user, "<span class='notice'>You fit [I] onto [src].</span>")
+				build_step++
+				qdel(I)
+			if(I.tool_behaviour == TOOL_CROWBAR) //Deconstruct
+				if(I.use_tool(src, user, 0, volume=30))
+					var/obj/item/mod/construction/helmet/newhelm = new(drop_location())
+					to_chat(user, "<span class='notice'>You pry [newhelm] from [src].</span>")
+					build_step--
+
+		if(6)
+			if(istype(I, /obj/item/mod/construction/boots)) //Construct
+				if(!user.temporarilyRemoveItemFromInventory(I))
+					return
+				playsound(src, 'sound/machines/click.ogg', 30, TRUE)
+				to_chat(user, "<span class='notice'>You fit [I] onto [src].</span>")
+				build_step++
+				qdel(I)
+			if(I.tool_behaviour == TOOL_CROWBAR) //Deconstruct
+				if(I.use_tool(src, user, 0, volume=30))
+					var/obj/item/mod/construction/gauntlets/newglove = new(drop_location())
+					to_chat(user, "<span class='notice'>You pry [newglove] from [src].</span>")
+					build_step--
+
+		if(7)
+			if(I.tool_behaviour == TOOL_WRENCH) //Construct
+				if(I.use_tool(src, user, 0, volume=30))
+					to_chat(user, "<span class='notice'>You wrench together the assembly.</span>")
+					build_step++
+					return
+			if(I.tool_behaviour == TOOL_CROWBAR) //Deconstruct
+				if(I.use_tool(src, user, 0, volume=30))
+					var/obj/item/mod/construction/boots/newboots = new(drop_location())
+					to_chat(user, "<span class='notice'>You pry [newboots] from [src].</span>")
+					build_step--
+
+		if(8)
+			if(I.tool_behaviour == TOOL_SCREWDRIVER) //Construct
+				if(I.use_tool(src, user, 0, volume=30))
+					to_chat(user, "<span class='notice'>You screw together the assembly.</span>")
+					build_step++
+					return
+			if(I.tool_behaviour == TOOL_WRENCH) //Deconstruct
+				if(I.use_tool(src, user, 0, volume=30))
+					to_chat(user, "<span class='notice'>You unwrench the assembled parts.</span>")
+					build_step--
+
+		if(9)
+			if(istype(I, /obj/item/mod/construction/armor)) //Construct
+				var/obj/item/mod/construction/armor/external_armor = I
+				if(!user.temporarilyRemoveItemFromInventory(external_armor))
+					return
+				playsound(src, 'sound/machines/click.ogg', 30, TRUE)
+				to_chat(user, "<span class='notice'>You fit [external_armor] onto [src], finishing your MODsuit.</span>")
+				qdel(external_armor)
+				qdel(src)
+				var/obj/item/modsuit = new /obj/item/mod/control(drop_location(), external_armor.theme)
+				user.put_in_hands(modsuit)
+			if(I.tool_behaviour == TOOL_SCREWDRIVER) //Construct
+				if(I.use_tool(src, user, 0, volume=30))
+					to_chat(user, "<span class='notice'>You unscrew the assembled parts.</span>")
+					build_step--
+	update_icon_state()
+
+/obj/item/mod/construction/shell/update_icon_state()
+	. = ..()
+	if(build_step == 1)
+		icon_state = "mod-construction"
```

Comment:
why not just name this "mod-construction[1]" so that this if statement won't be needed?